src/main/resources/templates/layouts/main.html

```diff
@@ -0,0 +1,25 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
+
+    <title data-layout-title-pattern="$LAYOUT_TITLE :: $CONTENT_TITLE">Plugin Health Scoring</title>
+
+    <link data-th-href="@{/style/index.css}" rel="stylesheet"/>
+
+    <script src="http://localhost:35729/livereload.js"
+            data-th-if="${#arrays.contains(@environment.getActiveProfiles(), 'dev')}"></script>
+</head>
+<body>
+
+<div id="app">
+    <header>
+        <span class="app--title">Plugin Health Scoring</span>
+    </header>
+    <main data-layout-fragment="content">
```

Comment:
I was wondering how this main page is showing details of the listing.html file, I guess this `<main>` tag with attribute `data-layout-fragment="content"` is rendering values of the listing.html file here on this page because that tag contains the rendering logic of each probe.

Reply:
this is only because of the `data-layout-fragment` attribute. This is coming from https://ultraq.github.io/thymeleaf-layout-dialect/processors/decorate/. I could have only use `data-th-fragment` from plain Thymeleaf templating as well: https://www.thymeleaf.org/doc/tutorials/3.0/usingthymeleaf.html#template-layout.

Reply:
I'm able to make more sense now after reading the first link you shared. This main.html layout acts as a sort of ‘default’, with our probe-listing content acting as implementations (which are defined inside of the fragment we've created) on top of this default.